### PR TITLE
remove random unneccessary files + bundle update

### PR DESCRIPTION
**Pull request by steved**

### CODE REVIEW

**Comment by steved** [comment]
@zendesk/samson

File: test/integration/cleanliness_test.rb [hunk 1]
```diff
@@ -11,14 +11,4 @@
     # if it is not avoidable see http://stackoverflow.com/questions/7163264
     File.read('Gemfile.lock').wont_include 'rails-assets-bootstrap '
   end
-
-  if ActiveRecord::Base.connection.adapter_name == "Mysql2"
-    it "uses the right row format in mysql" do
```
Comment (sandlerr) [inline test/integration/cleanliness_test.rb]: why remove this?

Comment (steved) [inline test/integration/cleanliness_test.rb]: I don't really understand what it's testing, the travis config? People can run samson without utf8mb4, so why do we care? I'd argue the monkey patch is also unnecessary and should be optionally included by users.

Comment (sandlerr) [inline test/integration/cleanliness_test.rb]: you're right, if utf8mb4 is not used then the index size limits won't be exceeded. It did take us a while to figure out why migrations were failing and how to fix it though. Is it reasonable to use the monkey patch in the test environment and keep this spec, but make it conditional on an envar?

Comment (steved) [inline test/integration/cleanliness_test.rb]: ok, added

Comment (steved) [inline test/integration/cleanliness_test.rb]: I'm pretty sure we're still using utf8 in our deployment too.

File: config/initializers/active_record.rb [hunk 2]
```diff
@@ -2,12 +2,17 @@
 # We also want our mysql databases to use utf8_mb4 encoding (the real unicode)
 # This monkey patch is from the following Github comment:
 # https://github.com/rails/rails/issues/9855#issuecomment-57665404
-if Gem.loaded_specs.has_key? 'mysql2'
-  require 'active_record/connection_adapters/mysql2_adapter'
+config = ActiveRecord::Base.configurations[Rails.env]
+
+if ENV.fetch('USE_UTF8M4', Rails.env.test?) && config['adapter'] == 'mysql2'
```
Comment (sandlerr) [inline config/initializers/active_record.rb]: nitpick: insert the B before the `4`

**Comment by sandlerr** [comment]
:+1:

File: config/initializers/active_record.rb (hunk 2 above)
Comment (steved) [inline config/initializers/active_record.rb]: whoops, thanks

File: test/integration/cleanliness_test.rb (hunk 1 above)
Comment (sandlerr) [inline test/integration/cleanliness_test.rb]: Yes, our deployment uses Compact

**Pull request merged**